charts/datahub/subcharts/datahub-gms/templates/deployment.yaml

```diff
@@ -311,6 +311,10 @@ spec:
                 secretKeyRef:
                   name: {{ .Values.global.datahub.metadata_service_authentication.systemClientSecret.secretRef }}
                   key: {{ .Values.global.datahub.metadata_service_authentication.systemClientSecret.secretKey }}
+            {{- if .Values.global.datahub.metadata_service_authentication.restApi.authorization.enabled }}
```

Comment:
Suggest:

```
- name: REST_API_AUTHORIZATION_ENABLED
  value: {{  .Values.global.datahub.metadata_service_authentication.restApi.authorization.enabled | quote }}
````

Reply:
Do we really want to set the value of the environment variable to the value of the config or do we just want to add the environment variable with the value true in case the config is true? I think it's more or less exactly the same logic here as for the global.datahub.metadata_service_authentication.view.authorization.enabled config?

https://github.com/acryldata/datahub-helm/blob/362671f26a417e0e0b4c3da7714ccd00f019cfb1/charts/datahub/subcharts/datahub-gms/templates/deployment.yaml#L314-L321

Reply:
@Masterchen09 - in this case, yes because we're going to change the application default [here](https://github.com/datahub-project/datahub/pull/11484) and mark it as a breaking change. When that change hits a release, will update the helm chart like this PR, however will follow @darnaut suggestion.

Reply:
@david-leifker This explanation makes sense (at least now after changing the default value 😉) - I have adjusted the PR accordingly and also set the default value to true to match the new default value (or should we wait with changing the default value in the chart, to be able to merge this PR directly before a new release is available?).

I have also checked the other (boolean) environment variables in all subcharts...there are more locations where the default value could currently not be overwritten because of the same reason - I have also adjusted these locations...maaaybe all boolean environment variables independent from the default value should be set using the same pattern, to be independent from changing default values in the future?

As I have nearly touched all subcharts, I have also fixed some formatting "issues" (some whitespaces...nothing special), adjusted the appVersion of each subchart to match the image tags resp. the version in the parent chart (would be great if the appVersion would always be matching the version in the parent chart or remove the appVersion from the subcharts...?) and removed an unused env.JMXPORT config from the value.yaml of the datahub-frontend, datahub-mae-consumer and datahub-mce-consumer subcharts (not sure why it was there). I hope it is fine to have this as part of this PR...somehow I just started to cleanup and could not stop it...😁

Reply:
@Masterchen09 - these look like good changes. Thank you once again for your contribution!